templates/apache-fragment.epp

```diff
@@ -0,0 +1,60 @@
+<%- |
+  Array[Struct[{
+    provider => Enum['directory', 'directorymatch', 'location', 'locationmatch', 'files', 'filesmatch', 'proxy', 'proxymatch'],
+    path => String[1],
+    Optional[options] => Array[String[1]],
+    Optional[allow_override] => Array[String[1]],
+    Optional[request_headers] => Array[String[1]],
+    Optional[proxy_pass] => Array[Struct[{
+      url => String[1],
+      Optional[params] => Hash[String, Variant[String, Integer]],
+      Optional[reverse_urls] => Array[String[1]],
+    }]],
+  }]] $directories,
+  Optional[Array[Hash[String, String]]] $proxy_pass = undef,
+| -%>
+<%-
+$directories.each |$directory| {
+  if $directory['provider'] =~ /^(.*)match$/ {
+    $provider = $1.capitalize + 'Match'
+  } else {
+    $provider = $directory['provider'].capitalize
+  }
+-%>
```

Comment:
This is hard to parse in this in this syntax. I think it could benefit from:

```
<%- $directories.each |$directory| { -%>
  <%- if $directory['provider'] =~ /^(.*)match$/ {
      $provider = $1.capitalize + 'Match'
    } else {
      $provider = $directory['provider'].capitalize
    }
  -%>
```